addons/hr_fleet/controllers/main.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import base64
+import io
+
+from odoo import http, _
+
+from PyPDF2 import  PdfFileReader, PdfFileWriter
+from reportlab.pdfgen import canvas
+
+
+class HrFleet(http.Controller):
+    @http.route(["/hrfleet/claimreport/<int:employee_id>"], type='http', auth='user')
+    def get_claim_report_user(self, employee_id, **post):
+
+        user_rec = http.request.env['res.users'].sudo().search([('id', '=', http.request.session.uid)])
+        if not user_rec.has_group('fleet.fleet_group_manager'):
+            return http.request.not_found()
+
+        Fleet = http.request.env['fleet.vehicle.assignation.log']
+        Attachment = http.request.env['ir.attachment']
+        Employee = http.request.env['hr.employee']
+
+        employee = Employee.search([('id','=',employee_id)],limit=1)
+        if not employee:
+            return http.request.not_found()
+
+        partner_id = employee.user_id.partner_id.id
+        if not partner_id:
+            return http.request.not_found()
+
+        car_line = Fleet.search([('driver_id','=',partner_id)])
+        doc_list = Attachment.search([('res_model', '=', 'fleet.vehicle.assignation.log'),
+                                    ('res_id', 'in', car_line.mapped('id'))],
+                                    order='create_date')
+
+        writer = PdfFileWriter()
+
+        font = "Helvetica"
+        normalFontSize = 14
+
+        for document in doc_list:
+
+            car_line_doc = Fleet.search([('driver_id','=',partner_id),("id","=",document.res_id)])
+            try:
+                reader = PdfFileReader(io.BytesIO(base64.b64decode(document.datas)), strict=False, overwriteWarnings=False)
+            except Exception:
+                continue
+
+            width = float(reader.getPage(0).mediaBox.getUpperRight_x())
+            height = float(reader.getPage(0).mediaBox.getUpperRight_y())
+
+            header = io.BytesIO()
+            can = canvas.Canvas(header)
+            can.setFont(font, normalFontSize)
+            can.setFillColorRGB(1,0,0)
+
+            car_name = car_line_doc[0].vehicle_id.display_name
+            date_start = str(car_line_doc[0].date_start)
+            date_end = str(car_line_doc[0].date_end)
+            if date_end == 'False':
+                date_end = '...'
+
+            text_to_print = "%s (drove from: %s to %s)" % (car_name, date_start, date_end)
```

Comment:
No need to use ``str`` as you use ``%s`` afterward.

Could be
``` python
            car_name = car_line_doc.vehicle_id.display_name
            date_start = car_line_doc.date_start
            date_end = car_line_doc.date_end or '...'
            text_to_print = _("%s (drove from: %s to %s)" % (car_name, date_start, date_end))
```
Also, shouldn't it be 'driven from' instead of 'drove from'?